src/python/tools/predict.py

```diff
@@ -0,0 +1,102 @@
+from pathlib import Path
+import argparse
+import numpy as np
+import torch
+
+
+def get_emb_dict(mapping_file, embs_file):
+    mapping = np.loadtxt(mapping_file, dtype=str, delimiter='\t')
+    mapping_dict = dict(mapping)
+    mapping_r_dict = dict(zip(mapping[:,1], mapping[:,0]))
+
+    num = len(mapping_dict)
+    embs = np.fromfile(embs_file, np.float32).reshape((num, -1))
+
+    return mapping_dict, mapping_r_dict, embs
+
+
+def perform_link_prediction(data_dir, dataset_dir, node, rel, rel_type, k):
+    node_mapping_file = Path(dataset_dir) / Path("node_mapping.txt")
+    rel_mapping_file = Path(dataset_dir) / Path("rel_mapping.txt")
+    node_embs_file = Path(data_dir) / Path("marius/embeddings/embeddings.bin")
+    lhs_embs_file = Path(data_dir) / Path("marius/relations/src_relations.bin")
+    rhs_embs_file = Path(data_dir) / Path("marius/relations/dst_relations.bin")
+
+    node_mapping_dict, node_mapping_r_dict, node_embs = get_emb_dict(node_mapping_file, node_embs_file)
+
+    if rel_type == 'lhs':
+        rel_mapping_dict, rel_mapping_r_dict, rel_embs = get_emb_dict(rel_mapping_file, lhs_embs_file)
+    else:
+        rel_mapping_dict, rel_mapping_r_dict, rel_embs = get_emb_dict(rel_mapping_file, rhs_embs_file)
+
+    node_id = int(node_mapping_dict.get(node))
+    rel_id = int(rel_mapping_dict.get(rel))
+    ctx_emb = node_embs[node_id] * rel_embs[rel_id]
```

Comment:
Looks like this is hardcoded to use DistMult. Add support for different models, I would recommend using the Python API of Marius and import the necessary relation operator and comparator.